aea/helpers/dependency_tree.py

```diff
@@ -192,7 +193,16 @@ def generate(cls, packages_dir: Path) -> List[List[PackageId]]:
         """Returns PublicId to hash mapping."""
         package_to_dependency_mappings = {}
         for component_type, component_file in COMPONENTS:
-            for file_path in packages_dir.glob(f"**/{component_file}"):
+            packages_list = []
+            if (packages_dir / VENDOR).exists():
+                packages_list = [
+                    *Path(".", VENDOR).glob(f"**/{component_file}"),
+                    *Path(".", f"{component_type}s").glob(f"**/{component_file}"),
+                ]
```

Comment:
When scanning for packages in an aea project, we only scan in `skills, connections, protocols, contracts` and `vendor` rather then scanning for whole directory.